File: AirCharter.API/AirCharter.API/Responses/Flights/PlaneCatalogResponse.cs
using AirCharter.API.Responses.Airports;

namespace AirCharter.API.Responses.Flights;

public sealed class PlaneCatalogResponse
{
    public int Id { get; set; }

    public string ModelName { get; set; } = string.Empty;

    public int PassengerCapacity { get; set; }

    public int MaxDistance { get; set; }

    public bool IsRouteFound { get; set; }

    public int DistanceKm { get; set; }

    public TimeSpan FlightTime { get; set; }

    public decimal FlightCost { get; set; }

    public int NumberOfTransfers { get; set; }

    public IReadOnlyCollection<AirportSearchResponse> RouteAirports { get; set; } =
        Array.Empty<AirportSearchResponse>();

    public IReadOnlyCollection<RouteLegResponse> RouteLegs { get; set; } =
        Array.Empty<RouteLegResponse>();

    public string? ImageBase64 { get; set; }
}

File: Frontend/src/contracts/responses/planes/planeCatalogResponse.ts
export interface PlaneCatalogResponse {
    id: number;
    modelName: string;
    passengerCapacity: number;
    maxDistance: number;
    isRouteFound: boolean;
    distanceKm: number;
    flightTime: string;
    flightCost: number;
    numberOfTransfers: number;
    routeAirports: AirportResponse[];
    routeLegs: RouteLegResponse[];
    imageBase64: string;
}

export interface AirportResponse {
    id: number;
    name: string;
    city: string;
    country: string;
    iata: string | null;
    icao: string;
    latitude: number;
    longitude: number;
}

export interface RouteLegResponse {
    fromAirportId: number;
    toAirportId: number;
    distanceKm: number;
    flightTime: string;
    flightCost: number;
    groundTimeAfterArrival: string | null;
}
